sources/api/bootstrap-containers/src/main.rs

```diff
@@ -0,0 +1,661 @@
+        }
+    }
+
+    match subcommand.as_deref() {
+        // Default subcommand is 'create-containers'
+        None | Some("create-containers") => Ok((global_args, Subcommand::CreateContainers {})),
+        Some("mark-bootstrap") => Ok((global_args, parse_mark_bootstrap_args(subcommand_args)?)),
+        Some(x) => {
+            return error::Usage {
+                message: format!("Unknown subcommand '{}'", x),
+            }
+            .fail()
+        }
+    }
+}
+
+/// Parses arguments for the 'mark-boostrap' subcommand
+fn parse_mark_bootstrap_args(args: Vec<String>) -> Result<Subcommand> {
+    let mut container_id = None;
+    let mut mode = None;
+
+    let mut iter = args.into_iter();
+    while let Some(arg) = iter.next() {
+        match arg.as_ref() {
+            "--container-id" => {
+                container_id = Some(iter.next().context(error::Usage {
+                    message: "Did not give argument to --container-id",
+                })?);
+            }
+
+            "--mode" => {
+                mode = Some(iter.next().context(error::Usage {
+                    message: "Did not give argument to --mode",
+                })?);
+            }
+
+            x => {
+                return error::Usage {
+                    message: format!("Unexpected argument '{}'", x),
+                }
+                .fail()
+            }
+        }
+    }
+
+    let container_id = container_id.context(error::Usage {
+        message: format!("Did not give argument to --container-id"),
+    })?;
+
+    let mode = mode.context(error::Usage {
+        message: format!("Did not give argument to --mode"),
+    })?;
+
+    Ok(Subcommand::MarkBootstrap(MarkBootstrapArgs {
+        container_id: container_id,
+        // Fail if 'mode' is invalid
+        mode: ValidContainerMode::try_from(mode.to_string())
+            .context(error::BootstrapContainerMode { mode })?,
+    }))
+}
+
+/// Handles how the bootstrap containers' systemd units are created
+fn handle_bootstrap_container<S>(
+    name: S,
+    container_details: &model::BootstrapContainer,
+) -> Result<()>
+where
+    S: AsRef<str>,
+{
+    let name = name.as_ref();
+
+    info!("Handling bootstrap container '{}'", name);
+
+    // Get basic settings, as retrieved from API.
+    let source = container_details
+        .source
+        .as_ref()
+        .context(error::MissingField {
+            name,
+            field: "source",
+        })?;
+
+    let mode = container_details
+        .mode
+        .as_ref()
+        .context(error::MissingField {
+            name,
+            field: "mode",
+        })?;
+
+    // If user data was specified, decode it and write it out
+    if let Some(user_data) = &container_details.user_data {
+        debug!("Decoding user data for container '{}'", name);
+        let decoded_bytes = base64::decode(user_data.as_bytes()).context(error::Base64Decode {
+            base64_string: user_data.as_ref(),
```

Comment:
Seems like that user_data could be quite large, do we want all of it in the error message?

Reply:
This is what we do in [host-containers](https://github.com/bottlerocket-os/bottlerocket/blob/e04cdf50b21661e0e1633420f4b6381c920a71cc/sources/api/host-containers/src/main.rs#L358), do you have a different  approach in mind?

Reply:
OK, seems a little untidy but not serious (unless it were seriously huge). An idea would be to give all of it if less than N bytes or MAX_BYTES if it is larger. Not worth blocking the feature for though!